### PR TITLE
Add a button in the controller config dialog to refresh the usb gc adapter status

### PR DESCRIPTION
**Pull request by mathieui**
This is useful when:
- The adapter was not plugged in when dolphin started
- The adapter gets unplugged (or unplugged and then re-plugged) while dolphin is running

Initially that also permitted to re-scan for the adapter when dolphin was terminated abruptly (like a segfault) and was unable to get a hold of the device quickly after startup, but this branch also fixes that.

### CODE REVIEW

**Comment by delroth** [comment]
Could we just have a periodic timer for this instead of introducing new UI elements?

File: Source/Core/Core/HW/SI_GCAdapter.cpp [hunk 1]
```diff
@@ -128,6 +128,7 @@ void Init()
 				u8 bus = libusb_get_bus_number(device);
 				u8 port = libusb_get_device_address(device);
 				ret = libusb_open(device, &s_handle);
+				bool continue_setup = true;
```
Comment (delroth) [inline Source/Core/Core/HW/SI_GCAdapter.cpp]: Looks like some code that would benefit from some goto magic.

**Comment by mathieui** [comment]
A periodic timer could be enough, yes, but I'm not too keen about changing configuration in the background, without the user knowing.

File: Source/Core/DolphinWX/ControllerConfigDiag.cpp [hunk 2]
```diff
@@ -164,22 +168,30 @@ wxStaticBoxSizer* ControllerConfigDiag::CreateGamecubeSizer()
...
-			gamecube_adapter->SetLabelText(_("Adapter Not Detected"));
-		gamecube_adapter->SetValue(false);
-		gamecube_adapter->Disable();
-		gamecube_adapter_thread->SetValue(false);
-		gamecube_adapter_thread->Disable();
+		{
+			m_gamecube_adapter->SetLabelText(_("Adapter Not Detected"));
+		}
+		m_gamecube_adapter->SetValue(false);
+		m_gamecube_adapter->SetValue(false);
```
Comment (skidau) [inline Source/Core/DolphinWX/ControllerConfigDiag.cpp]: Duplicate line here

File: Source/Core/DolphinWX/ControllerConfigDiag.h [hunk 3]
```diff
@@ -69,6 +72,30 @@ class ControllerConfigDiag : public wxDialog
 		SConfig::GetInstance().m_GameCubeAdapterThread = event.IsChecked();
 		event.Skip();
 	}
+	void OnGameCubeAdapterScan(wxCommandEvent& event)
+	{
+		SConfig::GetInstance().m_GameCubeAdapter = true;
+		SConfig::GetInstance().m_GameCubeAdapterThread = true;
```
Comment (skidau) [inline Source/Core/DolphinWX/ControllerConfigDiag.h]: We shouldn't be force enabling the thread.  It should be based on the user setting.

File: Source/Core/DolphinWX/ControllerConfigDiag.cpp (hunk 2 above)
Comment (Sonicadvance1) [inline Source/Core/DolphinWX/ControllerConfigDiag.cpp]: Really /really/ make sure it is set to false.

**Comment by skidau** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/HW/SI_GCAdapter.cpp [hunk 4]
```diff
@@ -177,6 +181,7 @@ void Init()
 				}
 				else
 				{
+continue_setup:
```
Comment (lioncash) [inline Source/Core/Core/HW/SI_GCAdapter.cpp]: Can this code not be extracted out into its own function?

**Comment by skidau** [comment]
@dolphin-emu-bot rebuild

**Comment by kkkllleee** [comment]
Why not have a window appear, telling you your controller has been unplugged and have the refresh button there? The only issue I see then, is that we may need a constant check, but if we place an event instead, then there shouldn't be an issue

**Comment by mathieui** [comment]
I don’t think anymore that this PR is useful, as the other adapter issues have been solved, and restarting dolphin is not a huge chore anyway.

**Pull request closed without merging**